src/VCAuthn/IdentityServer/StartupExtensions.cs

```diff
@@ -0,0 +1,87 @@
+
+            // Register identity server
+            services.AddIdentityServer(options =>
+                {
+                    options.Events.RaiseErrorEvents = true;
+                    options.Events.RaiseInformationEvents = true;
+                    options.Events.RaiseFailureEvents = true;
+                    options.Events.RaiseSuccessEvents = true;
+
+                    // When forwarded headers aren't sufficient we leverage PublicOrigin option
+                    options.PublicOrigin = config.GetSection("PublicOrigin").Value;
+                })
+                .AddConfigurationStore(options =>
+                {
+                    options.ConfigureDbContext = b =>
+                        b.UseNpgsql(connectionString, sql => sql.MigrationsAssembly(migrationsAssembly));
+                })
+                .AddOperationalStore(options =>
+                {
+                    options.ConfigureDbContext = b =>
+                        b.UseNpgsql(connectionString, sql => sql.MigrationsAssembly(migrationsAssembly));
+                    // this enables automatic token cleanup. this is optional.
+                    options.EnableTokenCleanup = true;
+                })
+
+                
+                // If cert supplied will parse and call AddSigningCredential(), if not found will create a temp one
+                .AddDeveloperSigningCredential(true, config.GetSection("CertificateFilename").Value)
+                ;
+        }
+        
+        public static void UseAuthServer(this IApplicationBuilder app, IConfiguration config)
+        {
+            InitializeDatabase(app, config.GetSection("RootClientSecret").Value);
+            app.UseIdentityServer();
+        }
+        
+         public static void InitializeDatabase(IApplicationBuilder app, string rootClientSecret)
+        {
+            using (var serviceScope = app.ApplicationServices.GetService<IServiceScopeFactory>().CreateScope())
+            {
+                //Resolve the required services
+                var configContext = serviceScope.ServiceProvider.GetRequiredService<ConfigurationDbContext>();
+                serviceScope.ServiceProvider.GetRequiredService<PersistedGrantDbContext>().Database.Migrate();
+
+                //Migrate any required db contexts
+                configContext.Database.Migrate();
+                
+                var currentIdentityResources = configContext.IdentityResources.ToList();
+                foreach (var resource in Config.GetIdentityResources())
+                {
+                    if (!currentIdentityResources.Any(_ => resource.Name == _.Name))
+                    {
+                        configContext.IdentityResources.Add(resource.ToEntity());
+                    }
+                }
+                configContext.SaveChanges();
+                
+                foreach (var client in Config.GetClients())
+                {
+                    configContext.Clients.Add(client.ToEntity());
```

Comment:
Do we need to do a similar check to the above, make sure the client doesn't already exist on repeat intializations

Reply:
E.g check if the client exists based on name if so either update the entity or skip, otherwise if it doesn't exist insert.

Reply:
👍 
good idea